*: add initial read benchmark for etcd v3

yichengq opened the pull request:
It includes the initial read benchmark for etcd v3.

This is the first step to give some rough thoughts. I haven't digged
deeper to answer some questions, including why its performance is not
better than HTTP + json, why one put will cause performance downgrade.

for #3290

xiang90 commented on hack/v3benchmark/main.go:
```diff
@@ -0,0 +1,87 @@
+	"github.com/rakyll/pb"
+)
+
+func main() {
```
can we move this to /tools?

xiang90 replied:
Also I would suggest to check https://github.com/grpc/grpc-go/blob/master/benchmark/benchmark.go

yichengq commented:
Moved.

I checked the benchmark written by gRPC, and finds that it uses one connection for all requests in concurrent way, while I use multiple connections, each of which sends one request at one time. I have added a TODO in code to support that.

Having played with there two methods, I find using one connection could perform 10% better when having 256 clients, and performs nearly the same in other situations. There is still gap between its performance and expectation. The benchmark today cannot match the grpc internal benchmark, which will be researched soon. 

For etcd, i would expect the use case is O(100-1000) connections with O(1) concurrency in each connection.

xiang90 commented:
We can merge as it is today. I just briefly look through the code and did not give it a careful read.

xiang90 commented on tools/v3benchmark/main.go:
```diff
@@ -0,0 +1,88 @@
+package main
```
add coreos licensee

yichengq commented:
license added.

It performs the same way as boom does, and nearly the same way as grpc-benchmark does except the problem that how many concurrency on one connection.

xiang90 commented:
Let merge this as it is. This would be a good start.

yichengq commented:
:ship:
